src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/MsaPassthroughHandler.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Requests;
+using Windows.Security.Credentials;
+
+namespace Microsoft.Identity.Client.Platforms.Features.WamBroker
+{
+    internal class MsaPassthroughHandler : IMsaPassthroughHandler
+    {
+        public const string TransferTokenScopes =
+            "openid profile offline_access service::http://Passport.NET/purpose::PURPOSE_AAD_WAM_TRANSFER";
+        private readonly ICoreLogger _logger;
+        private readonly IWamPlugin _msaPlugin;
+        private readonly IWamProxy _wamProxy;
+        private readonly IntPtr _parentHandle;
+
+        public MsaPassthroughHandler(
+            ICoreLogger logger,
+            IWamPlugin msaPlugin,
+            IWamProxy wamProxy,
+            IntPtr parentHandle)
+        {
+            _logger = logger;
+            _msaPlugin = msaPlugin;
+            _wamProxy = wamProxy;
+            _parentHandle = parentHandle;
+        }
+
+        public bool IsPassthroughEnabled(AuthenticationRequestParameters authenticationRequestParameters)
+        {
+            // TODO: change this to a config object
```

Comment:
how do these extraqp's get set? does the developer set this? I didn't see this in the manual tests. how does msal know it's msa-pt?

Reply:
Our (me + JM) initial thought was for developers to tell us that the app is MSA-PT via extraqueryparams. Note that MSA-PT is a legacy 1st party only "feature", so we don't want to polute our API surface with it. 

After some thought, I realized that we might need this at the AppConfig level, so I'll add an actual API for it, smth like `WithMsaPassthrough(true)` but hide it in an obscure namespace. This will be a different PR.

Reply:
idk about adding another API for this small use case. I would just keep the extra qp and document it. agree on not polluting the API surface, especially w/more 1st party only APIs.

---
In reply to: [565513741](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2365#discussion_r565513741) [](ancestors = 565513741)